### PR TITLE
Gian refactor search reducer

### PR DESCRIPTION
**Pull request by GPaoloni**
This PR refactors search reducer in order to match the new structure for our reducers.

### CODE REVIEW

File: plugin-hrm-form/src/states/ActionTypes.js [hunk 1]
```diff
@@ -5,21 +5,12 @@ export const HANDLE_CHANGE = 'HANDLE_CHANGE';
 export const SAVE_END_MILLIS = 'SAVE_END_MILLIS';
 export const CHANGE_TAB = 'CHANGE_TAB';
 export const RESTORE_ENTIRE_FORM = 'RESTORE_ENTIRE_FORM';
+export const HANDLE_SELECT_SEARCH_RESULT = 'HANDLE_SELECT_SEARCH_RESULT';
```
Comment (GPaoloni) [inline plugin-hrm-form/src/states/ActionTypes.js:8]: I let the name of this as it is, but I think we might want to rename it, to make clear that this actually does a "copy contact information".

File: plugin-hrm-form/src/states/ContactState.js [hunk 2]
```diff
@@ -79,6 +79,12 @@ export class Actions {
   static prepopulateFormCaller = (gender, age, taskId) => ({ type: PREPOPULATE_FORM_CALLER, gender, age, taskId });
 }
 
+export const handleSelectSearchResult = (searchResult, taskId) => ({
```
Comment (GPaoloni) [inline plugin-hrm-form/src/states/ContactState.js:82]: Same comment as above.

File: plugin-hrm-form/src/states/ContactState.js [hunk 3]
```diff
@@ -79,6 +79,12 @@ export class Actions {
...
+  type: HANDLE_SELECT_SEARCH_RESULT,
+  searchResult,
+  taskId,
+});
+
```
Comment (GPaoloni) [inline plugin-hrm-form/src/states/ContactState.js:87]: Same comment as above.

File: plugin-hrm-form/src/types/types.ts [hunk 4]
```diff
@@ -24,3 +24,64 @@ export type Case = {
...
+      callSummary: string;
+      referredTo: string;
+      status: string;
+      keepConfidential: boolean;
+      okForCaseWorkerToCall: boolean;
+      howDidTheChildHearAboutUs: string;
+      didYouDiscussRightsWithTheChild: boolean;
+      didTheChildFeelWeSolvedTheirProblem: boolean;
+      wouldTheChildRecommendUsToAFriend: boolean;
+    };
```
Comment (GPaoloni) [inline plugin-hrm-form/src/types/types.ts:66]: If we ever factor out the children and case forms as we did for caller information form, the type here will be much simple (like above callerInformation).

File: plugin-hrm-form/src/states/search/types.ts [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+};
+
+// this was inside search reducer, but it's used in ContactDetails and Timeline components. Should this be moved to a more common/shared place?
+export const ContactDetailsSections = {
+  GENERAL_DETAILS: 'General details',
+  CALLER_INFORMATION: 'Caller information',
+  CHILD_INFORMATION: 'Child information',
+  ISSUE_CATEGORIZATION: 'Issue categorization',
+  CONTACT_SUMMARY: 'Contact summary',
+} as const;
```
Comment (murilovmachado) [inline plugin-hrm-form/src/states/search/types.ts:33]: I agree we can find a more general place to put this type.

**Review by murilovmachado — APPROVED** [review]
Looks good to me!

**Pull request merged**